src/main/java/nl/hsac/fitnesse/junit/JUnitAllureFrameworkListener.java

```diff
@@ -0,0 +1,266 @@
+        AnnotationManager am = new AnnotationManager(description.getAnnotations());
+        am.update(event);
+        this.fireClearStepStorage();
+        this.getLifecycle().fire(event);
+
+        String fullTestName = description.getMethodName();
+        String nameParts[] = fullTestName.split("\\.");
+        String testName = nameParts[nameParts.length-1];
+        String tagInfo = null;
+        for(WikiPage page : pages){
+            if(testName.equalsIgnoreCase(page.getName())){
+                tagInfo = page.getData().getProperties().get("Suites");
+            }
+        }
+        if(tagInfo != null) {
+            createStories(tagInfo);
+        }
+    }
+
+    public void testFailure(Failure failure) {
+        if(failure.getDescription().isTest()) {
+            Throwable exception = failure.getException();
+            List<Pattern> patterns = new ArrayList<>();
+            patterns.add(SCREENSHOT_PATTERN);
+            patterns.add(PAGESOURCE_PATTERN);
+            processAttachments(exception, patterns);
+
+            this.fireTestCaseFailure(exception);
+            this.recordTestResult(failure.getDescription());
+
+        } else {
+            this.startFakeTestCase(failure.getDescription());
+            this.fireTestCaseFailure(failure.getException());
+            this.finishFakeTestCase();
+        }
+    }
+
+    public void testAssumptionFailure(Failure failure) {
+        this.testFailure(failure);
+    }
+
+    public void testIgnored(Description description) {
+        this.startFakeTestCase(description);
+        this.getLifecycle().fire((new TestCasePendingEvent()).withMessage(this.getIgnoredMessage(description)));
+        this.finishFakeTestCase();
+    }
+
+    public void testFinished(Description description) {
+        fitnesseResult(description.getMethodName());
+        this.getLifecycle().fire(new TestCaseFinishedEvent());
+    }
+
+    public void testSuiteFinished(String uid) {
+        this.getLifecycle().fire(new TestSuiteFinishedEvent(uid));
+    }
+
+    public void testRunFinished(Result result) {
+
+        for (String uid : this.getSuites().values()) {
+            this.testSuiteFinished(uid);
+        }
+
+    }
+
+    public String generateSuiteUid(String suiteName) {
+        String uid = UUID.randomUUID().toString();
+        synchronized(this.getSuites()) {
+            this.getSuites().put(suiteName, uid);
+            return uid;
+        }
+    }
+
+    public String getSuiteUid(Description description) {
+        String suiteName = description.getClassName();
+        if(!this.getSuites().containsKey(suiteName)) {
+            Description suiteDescription = Description.createSuiteDescription(description.getTestClass());
+            this.testSuiteStarted(suiteDescription);
+        }
+
+        return this.getSuites().get(suiteName);
+    }
+
+    public String getIgnoredMessage(Description description) {
+        Ignore ignore = description.getAnnotation(Ignore.class);
```

Comment:
Will this ever work for a FitNesse page? (I don't think they have `@Ignore` annotations)